src/html.rs

```diff
@@ -232,19 +233,51 @@ impl MsgId {
             .await;
 
         if let Some(rawmime) = rawmime {
-            match HtmlMsgParser::from_bytes(context, rawmime.as_bytes()).await {
-                Err(err) => format!("parser error: {}", err),
-                Ok(parser) => parser.html,
+            if !rawmime.is_empty() {
+                match HtmlMsgParser::from_bytes(context, rawmime.as_bytes()).await {
+                    Err(err) => {
+                        warn!(context, "get_html: parser error: {}", err);
+                        None
+                    }
+                    Ok(parser) => Some(parser.html),
+                }
+            } else {
+                warn!(context, "get_html: empty mime for {}", self);
+                None
             }
         } else {
-            format!("parser error: no mime for {}", self)
+            warn!(context, "get_html: no mime for {}", self);
+            None
         }
     }
+
+    /// Wraps HTML generated by get_html() into a text/html mimepart structure.
+    /// Used on forwarding messages to avoid leaking the original mime structure
```

Comment:
```suggestion
    /// Wraps HTML generated by [`MsgId::get_html`] into a text/html mimepart structure.
    ///
    /// Used on forwarding messages to avoid leaking the original mime structure
```